bgcval2/analysis_timeseries.py

```diff
@@ -88,6 +89,29 @@ def listModelDataFiles(jobID, filekey, datafolder, annual):
     return model_files
 
 
+def list_input_files(files_path, dictionary, paths):
+    """
+    Generate a list of files from a path, which may include 
+    several $PATH values.
+    """
+    #####
+    # Replace some values for $FLAGS in the path:
+    flags = ['USERNAME','basedir_model', 'basedir_obs','PATHS_GRIDFILE']
+    flag_values = [getpass.getuser(), paths.ModelFolder_pref, paths.ObsFolder, paths.orcaGridfn]
+
+    for flag in ['jobID', 'model', 'years','year', 'scenario']:
+        if dictionary.get(flag, False):
+            flags.append(flag.upper())
+            flag_values.append(dictionary[flag])
+ 
+    for flag, flag_value in zip(flags, flag_values):
+        print('Changing FLAG:',flag,'to',flag_value, 'in', files_path)
+        files_path = findReplaceFlag(files_path, flag, flag_value)
+
+    input_files = sorted(glob(files_path))
```

Comment:
I wonder if this glob should be done elsewhere, just to separate this into two functions. One to find/replace strings and the other to generate a list of files.

Reply:
I'll have a look now, bud

Reply:
looks to be in the right function, I think. If you want to go more general you can create a custom glob function that can be called from outside

Reply:
Okay, it could get weird later if lots of kwargs get added, but we'll burn that bridge when we get to it,

Reply:
dude, this https://www.youtube.com/watch?v=kgeIINs1TrQ :grin: